### PR TITLE
Avoid tangled implicit dependencies on absorption times

### PR DESCRIPTION
**Pull request by ps2**
This clears up some implicit dependencies that are non-obvious to change then default carb absorption is changed. 

https://tidepool.atlassian.net/browse/LOOP-492

### CODE REVIEW

File: LoopKit/TemporaryScheduleOverrideHistory.swift [hunk 1]
```diff
@@ -68,6 +68,8 @@ public final class TemporaryScheduleOverrideHistory {
     }
     
     private var modificationCounter: Int
+    
+    public var relevantTimeWindow: TimeInterval = TimeInterval.hours(8)
```
Comment (mpangburn) [inline LoopKit/TemporaryScheduleOverrideHistory.swift]: I see this gets set in Loop, so the default isn't really used, but do you think it's worth changing to 10 hours anyway?

**Review by mpangburn — APPROVED** [review]
LGTM!

**Review by darinkrauss — APPROVED** [review]
Nice, I like dropping the duplicate from the CarbStore. LGTM! 🎯

File: LoopKitTests/TemporaryScheduleOverrideHistoryTests.swift [hunk 2]
```diff
@@ -221,8 +221,8 @@ final class TemporaryScheduleOverrideHistoryTests: XCTestCase {
             RepeatingScheduleValue(startTime: .hours(0), value: 1.8),
             RepeatingScheduleValue(startTime: .hours(4), value: 1.2),
             RepeatingScheduleValue(startTime: .hours(6), value: 1.4),
-            RepeatingScheduleValue(startTime: .hours(20), value: 1.0),
-            RepeatingScheduleValue(startTime: .hours(22), value: 1.5),
+            RepeatingScheduleValue(startTime: .hours(20), value: 1.5),
+            //RepeatingScheduleValue(startTime: .hours(22), value: 1.5),
```
Comment (mpangburn) [inline LoopKitTests/TemporaryScheduleOverrideHistoryTests.swift]: The updated test looks right, so the commented out line here can be deleted

File: LoopKitTests/TemporaryScheduleOverrideHistoryTests.swift [hunk 3]
```diff
@@ -221,8 +221,8 @@ final class TemporaryScheduleOverrideHistoryTests: XCTestCase {
...
         ])!
 
         print(expected)
```
Comment (mpangburn) [inline LoopKitTests/TemporaryScheduleOverrideHistoryTests.swift]: (If you wouldn't mind, you can delete this old print statement too, I must've missed it when I wrote this originally!)

**Review by mpangburn — APPROVED** [review]
Test fixes LGTM. Thanks!

**Pull request merged**